python/smqtk/algorithms/descriptor_generator/caffe_descriptor.py

```diff
@@ -408,8 +408,9 @@ def _process_batch(self, uuids4proc, data_elements, descr_elements, procs):
         self._log.debug("extracting return layer '%s' into descriptors",
                         self.return_layer)
         for uid, v in zip(uuids4proc, self.network.blobs[self.return_layer].data):
-            descr_elements[uid].set_vector(v)
-
+            # In case caffe generates multidimensional array (rows, 1, 1)
+            # or matrix format result
+            descr_elements[uid].set_vector(numpy.array(v, copy=False).flatten())
```

Comment:
> if v.ndim > 1:
>     v = numpy.array(v, copy=False).flatten()
> descr_elements[uid].set_vector(v)
> ...

if that's the case, then does it makes sense to pass copy=False? Let me look at the docs